src/NodaTime/TimeZones/ZoneRecurrence.cs

```diff
@@ -129,27 +129,17 @@ public bool Equals(ZoneRecurrence other)
             else if (safeLocal >= maxLocalInstant)
             {
                 // Asked for a transition after our final transition... or both are beyond the end of time (in which case
-                // we can return an infinite transition)
-                return maxLocalInstant == LocalInstant.AfterMaxValue ? new Transition(Instant.AfterMaxValue, newOffset) :  (Transition?) null;
+                // we can return an infinite transition). THis branch will always be taken for transitions beyond the end
```

Comment:
Also, while you're here, there's an odd construction I can't parse on L22 with "before that calendars".

Reply:
Yup, will fix that and the typo, thanks.